### PR TITLE
Improve connectivity check

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Clearly relying on the `rtc connected` console log line appearing to decide whether to start the recording wasn't ideal. On top of that, the event could technically be fired multiple times which would have been fatal (closing the ready channel twice).

PR improves this a bit through a polling function that evaluates a proper JS expression to figure out when it's safe to begin.

### CODE REVIEW

File: cmd/recorder/recorder.go [hunk 1]
```diff
@@ -138,13 +149,21 @@ func (rec *Recorder) runBrowser(recURL string) error {
 
 	log.Printf("stop received, shutting down browser")
 
+	var disconnected bool
+	disconnectCheckExpr := "window.callsClient.disconnect(); !window.callsClient || window.callsClient.closed"
 	if err := chromedp.Run(ctx,
-		chromedp.EvaluateAsDevTools("window.callsClient.disconnect();", nil),
+		chromedp.Evaluate(disconnectCheckExpr, &disconnected),
```
Comment (cpoile) [inline cmd/recorder/recorder.go:155]: Just to make sure I understand, this evals the `disconnectCheckExpr`, and waits for the eval to finish (so it must be a blocking call, but without a timeout), and then if it returns false, we... do the same thing we do if it was successful?

**Review by cpoile — APPROVED** [review]
This is wild, I didn't know you could do this. Very cool.
Just have a non-blocking question for my own understanding.

File: cmd/recorder/recorder.go (hunk 1 above)
Comment (streamer45) [reply cmd/recorder/recorder.go:155]: Yes, it's really just for logging/debugging purposes as we are shutting down the process anyway but would prefer to do it cleanly if possible.

Comment (cpoile) [reply cmd/recorder/recorder.go:155]: Ah, ok, makes sense.

**Pull request merged**